src/app/clusters/identify-server/IdentifyCluster.cpp

```diff
@@ -0,0 +1,236 @@
+    mOnEffectIdentifier(config.onEffectIdentifier), mCurrentEffectIdentifier(config.effectIdentifier),
+    mEffectVariant(config.effectVariant),
+    mTimerDelegate(config.timerDelegate == nullptr ? &sDefaultTimerDelegate : config.timerDelegate)
+{}
+
+DataModel::ActionReturnStatus IdentifyCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                             AttributeValueEncoder & encoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::IdentifyTime::Id:
+        return encoder.Encode(mIdentifyTime);
+    case Attributes::IdentifyType::Id:
+        return encoder.Encode(mIdentifyType);
+    case Attributes::ClusterRevision::Id:
+        return encoder.Encode(Identify::kRevision);
+    case Attributes::FeatureMap::Id:
+        return encoder.Encode<uint32_t>(0);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+DataModel::ActionReturnStatus IdentifyCluster::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                              AttributeValueDecoder & decoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::IdentifyTime::Id: {
+        mPreviousIdentifyTime = mIdentifyTime;
+        ReturnErrorOnFailure(decoder.Decode(mIdentifyTime));
+        IdentifyTimeAttributeChanged(/* isWrittenByClientOrCmd */ true);
+        return CHIP_NO_ERROR;
+    }
+    break;
+    // Read-only attributes
+    case Attributes::IdentifyType::Id:
+        return Protocols::InteractionModel::Status::UnsupportedWrite;
+    case Attributes::ClusterRevision::Id:
+        return Protocols::InteractionModel::Status::UnsupportedWrite;
+    case Attributes::FeatureMap::Id:
+        return Protocols::InteractionModel::Status::UnsupportedWrite;
+    // Invalid attributes
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+CHIP_ERROR IdentifyCluster::Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    AttributeListBuilder listBuilder(builder);
+    return listBuilder.Append(Span(Identify::Attributes::kMandatoryMetadata), {});
+}
+
+void IdentifyCluster::TimerFired()
+{
+    if (mIdentifyTime > 0)
+    {
+        mPreviousIdentifyTime = mIdentifyTime;
+        mIdentifyTime--;
+        IdentifyTimeAttributeChanged(false);
+    }
+}
+
+// According to the spec, section 5.1 - IdentifyTime Attribute:
+// Changes to this attribute SHALL only be marked as reportable in the following cases:
+// 1. When it changes from 0 to any other value and vice versa, or
+// 2. When it is written by a client, or
+// 3. When the value is set by an Identify command.
+void IdentifyCluster::IdentifyTimeAttributeChanged(bool isWrittenByClientOrCmd)
+{
+    // Start Identify
+    if (mPreviousIdentifyTime == 0 && mIdentifyTime > 0)
```

Comment:
This method overall feels odd:

- seems like a post-processing function that uses mPreviousIdentifyTime, which is required to be stored by the caller. I would prefer this method being a `SetIdetifyTime` and it can store previous time as a local variable. Saves some minor ram and makes the code clearer without imposing prerequisites on the caller.

- Logic for changed or not could be encapsulating by returning kWriteSuccessNoOp (https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/ActionReturnStatus.h#L51) if we want to indicate a "success but do not modify". Unsure if this makes the code more readable, however a single place for calling NotifyAttributeChanged seems desirable.